### PR TITLE
flag to disable stdout capture during tests

### PR DESCRIPTION
**Pull request by bbatha**
I'm not entirely sure how to get this working in chrome.

I couldn't even normal chrome usage when testing. I attempted in windows and the WSL environment with Chrome 58.

### CODE REVIEW

**Comment by koute** [comment]
> I'm not entirely sure how to get this working in chrome.

You mean how to pass the `--nocapture`? I haven't tried it, but I'm guessing that we'd have to set `Module['arguments']` and pass the `--nocapture` there just as it's passed to a real tests executable. (If you look at the `DEFAULT_TEST_INDEX_HTML` you can see that I'm currently setting `print`, `printErr` and `onExit` on `Module` - that's where you'd set the arguments.)

> I couldn't even normal chrome usage when testing.

Do you mean that running tests through Chrome on Windows doesn't work at all? (Assuming your other PR is merged in.)

**Comment by bbatha** [comment]
I added `--nocapture` to `Module['arguments']`.

> Do you mean that running tests through Chrome on Windows doesn't work at all? (Assuming your other PR is merged in.)

Ya chrome launches but the tests don't. For native windows this makes sense, the chrome [dev blog](https://developers.google.com/web/updates/2017/04/headless-chrome) says that headless doesn't work on windows yet. In WSL I get a time out with "No such process", I'm not entirely sure if chrome is supposed to work in WSL though.

**Comment by koute** [comment]
> Ya chrome launches but the tests don't.

Hmm... this is strange. Even if your copy of Chrome doesn't support headless it should work - without headless support it'll just flash an empty page at you while it's running the tests (or at least that's how it works on Linux if you have Chrome which doesn't support headless).

Could you perhaps play with the the arguments which we're passing to Chrome and see if that might help? Also you could try to go into Chrome's developer tools and see if an exception was thrown or something. (If you replace the `--app=$URL` parameter with `$URL` it should run with the menu visible, I think.)

(On a side note the whole way we're running tests in Chrome right now is definitely suboptimal and I do plan to rework it sooner or later. Instead of doing it the way it's done right now I'd be better to connect to Chrome using the Chrome Debugging Protocol and control it through that; I haven't had the time to investigate this yet though.)

**Comment by bbatha** [comment]
I've been thinking about this. It may be better to add a pass through argument like cargo does so that not every flag that the test runner supports needs to be promoted to a cargo web flag. For instance, `cargo test -- --nocapture` could become `cargo web test -- --nocapture`. I couldn't find an obvious way to do this with `clap` though. I'll do some more digging though.

**Comment by bbatha** [comment]
Figured it out. The most recent commit will just pass through anything after `--` to the test harness.

File: Cargo.lock [hunk 1]
```diff
@@ -1263,6 +1302,7 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 "checksum serde_derive 1.0.7 (registry+https://github.com/rust-lang/crates.io-index)" = "0019cd5b9f0529a1a0e145a912e9a2d60c325c58f7f260fc36c71976e9d76aee"
 "checksum serde_derive_internals 0.15.0 (registry+https://github.com/rust-lang/crates.io-index)" = "021c338d22c7e30f957a6ab7e388cb6098499dda9fd4ba1661ee074ca7a180d1"
 "checksum serde_json 0.9.10 (registry+https://github.com/rust-lang/crates.io-index)" = "ad8bcf487be7d2e15d3d543f04312de991d631cfe1b43ea0ade69e6a8a5b16a1"
+"checksum serde_json 1.0.2 (registry+https://github.com/rust-lang/crates.io-index)" = "48b04779552e92037212c3615370f6bd57a40ebba7f20e554ff9f55e41a69a7b"
 "checksum sha1 0.2.0 (registry+https://github.com/rust-lang/crates.io-index)" = "cc30b1e1e8c40c121ca33b86c23308a090d19974ef001b4bf6e61fd1a0fb095c"
 "checksum sha2 0.5.2 (registry+https://github.com/rust-lang/crates.io-index)" = "84d24a04bbdb2b09512951fc4d13ac76bb309c63163a22394a92698183a31d67"
 "checksum shell32-sys 0.1.1 (registry+https://github.com/rust-lang/crates.io-index)" = "72f20b8f3c060374edb8046591ba28f62448c369ccbdc7b02075103fb3a9e38d"
```
Comment (koute) [inline Cargo.lock]: Please put the `Cargo.lock` changes in a separate commit.

File: src/main.rs [hunk 2]
```diff
@@ -157,13 +160,14 @@ const DEFAULT_TEST_INDEX_HTML: &'static str = "
         Module['print'] = __cargo_web.print;
         Module['printErr'] = __cargo_web.print;
         Module['onExit'] = __cargo_web.on_exit;
+        Module['arguments'] = [{{#each arguments}} "{{this}}", {{/each}}];
```
Comment (koute) [inline src/main.rs]: Unfortunately this seems broken? It emits `Module['arguments'] = [ "[object]", ];`

File: src/main.rs [hunk 3]
```diff
@@ -712,18 +718,26 @@ fn command_test< 'a >( matches: &clap::ArgMatches< 'a >, project: &CargoProject
                     return Err( Error::EnvironmentError( "node.js not found; please install it!".into() ) );
                 };
 
-            let status = Command::new( nodejs_name ).arg( artifact ).run();
+            let test_args = std::iter::once( artifact.as_os_str() )
+               .chain( arg_passthrough.iter().map(|x| *x) );
```
Comment (koute) [inline src/main.rs]: `.map(|x| *x)` -> `.cloned()`

**Comment by bbatha** [comment]
Requested changes made.

**Comment by koute** [comment]
Thanks!

**Pull request merged**